src/utils/cloud_storage.R

```diff
@@ -1,91 +1,138 @@
+#' @returns Nothing, but file is written to the `hdx-signals` bucket.
 #'
 #' @export
-update_gcs_file <- function(df, name) {
+update_az_file <- function(df, name) {
+  tf <- tempfile(fileext = ".parquet")
+
+  arrow$write_parquet(
+    x = df,
+    sink = tf
+  )
+
   if (gmas_test_run()) {
     message(
-      "`update_pq_file()` not saving data as `gmas_test_run()` is `TRUE`. ",
+      "`update_az_file()` not saving data as `gmas_test_run()` is `TRUE`. ",
       "Set `GMAS_TEST_RUN` env variable to `FALSE` if you want the data to be ",
       "saved, but be careful of sending emails or calling the OpenAI API."
     )
     return(invisible(NULL))
   }
 
-  arrow$write_parquet(
-    x = df,
-    sink = bucket$path(name)
+  invisible(
+    utils$capture.output(
+      az$upload_blob(
+        container = blob,
+        src = tf,
+        dest = name
+      )
+    )
   )
 }
 
-#' Find GCS file names matching pattern
+#' Find MADS file names matching pattern
 #'
-#' Pulls names from the `global-monitoring` bucket and
-#' filters those names by the passed patterns, if passed. If `pattern` is `NULL`,
-#' then all files in the bucket are returned.
+#' Pulls names of files (not folders) from the `hdx-signals` container and
+#' filters those names by the passed patterns, if provided If `pattern` is `NULL`,
+#' then all files in the container are returned.
 #'
 #' @param pattern Pattern to look for. Passed to [stringr::str_detect()]
 #'
 #' @export
-gcs_file_detect <- function(pattern = NULL) {
-  file_names <- bucket$ls(recursive = TRUE)
+az_file_detect <- function(pattern = NULL) {
+  # get blob files but don't return dirs
+  blob_df <- az$list_blobs(blob)
+  blob_df <- blob_df[!blob_df$isdir,]
+  file_names <- blob_df$name
   if (!is.null(pattern)) {
     file_names <- file_names[stringr$str_detect(file_names, pattern)]
   }
   file_names
 }
 
-######################
-#### SETUP BUCKET ####
-######################
+#########################
+#### SETUP CONTAINER ####
+#########################
+
+#' Get the endpoint URL
+#'
+#' Currently system uses the blob endpoint, but allows access to the file endpoint
+#' as well if necessary. Used to create blob object in `blob_endpoint`.
+#'
+#' @param service Service to access, either blob (default) or file.
+azure_endpoint_url <- function(service = c("blob", "file")) {
+  service <- rlang$arg_match(service)
+  endpoint <- glue$glue(Sys.getenv("HDX_SIGNALS_ENDPOINT"))
+}
+
+# gets the Dsci blob endpoint using the HDX Signals SAS
+blob_endpoint <- az$blob_endpoint(
+  endpoint = azure_endpoint_url("blob"),
+  sas = Sys.getenv("HDX_SIGNALS_SAS")
+)
 
-# this bucket used to read and write from the GCS parquet
-bucket <- arrow$gs_bucket(
-  bucket = "global-monitoring",
-  json_credentials = Sys.getenv("GLOBAL_MONITORING_JSON")
+# blob object for HDX Signals, used to read and write data
+blob <- az$blob_container(
+  endpoint = blob_endpoint,
+  name = "hdx-signals"
```

Comment:
For me, the workflow would be easier to follow if these two calls were just added at the top of each `update_*.R` scripts or perhaps loaded in a different way as this is script is all functions other than these last calls. I feel like running/sourcing a module/or script in an analysis script that actually loads/assigns objects in the the env feels a bit weird for R-users, but might be more common in other languages/production related work flows?

Reply:
Remember we are using `box::use()`. If we were using `source()`, these objects would be loaded into the environment. However, as is, this is just standard programming practice in a way that I am defining global objects that can be used across the different functions in this specific module. So users never use or even see these objects, but they are used within the `update_az_file()` and `read_az_file()` functions, which are called frequently in other modules.

So, I think adding this elsewhere would violate the do not repeat yourself rule.

Reply:
fair, I don't really have any issue with it as i'm familiar with the code. I was more thinking if someone new was troubleshooting down the road.. how would they know these objects are loaded and made here specifically? But yeah im sure your right and im just not used to seeing it like this.

Reply:
Yeah, but in this case, those objects, for instance `blob` are defined in that module `cloud_storage.R`, but they are never used outside of it! Only the exported functions are, which are clearly imported in other scripts, so I would hope it's clear to the user. I think I agree with your point about making sure anything `source()` is just functions, but `box::use()` ONLY imports things that have the `@export` roxygen tag, so takes care of this nicely I think!